src/lib/mongoBackend/mongoUpdateContextSubscription.cpp

```diff
@@ -231,6 +231,24 @@ HttpStatusCode mongoUpdateContextSubscription
     }
   }
 
+  /* Description */
+  if (requestP->descriptionProvided)
+  {
+    // Note that in the case of description "" the field is deleted
+    if (requestP->description != "")
+    {
+      newSub.append(CSUB_DESCRIPTION, requestP->description);
+    }
+  }
```

Comment:
First version used requestP->description comparation to "". However, the problem is that, from a parsing layer point of view, description equal "" correspond to two situations:

1. description field was included but with value ""
2. description field was not included

while the bevahiour is not the same, as it is respectively

1. delete the subscription field
2. don't touch description (i.e. keep the one in the original subscription, if any)

Please, have a look to this fragment after taking that into account, and let's see if now it makes sense...